pkg/tuning/preset-tuning.go

```diff
@@ -2,11 +2,142 @@ package tuning
 
 import (
 	"context"
+	"fmt"
+	"os"
+
 	kaitov1alpha1 "github.com/azure/kaito/api/v1alpha1"
 	"github.com/azure/kaito/pkg/model"
+	"github.com/azure/kaito/pkg/resources"
+	"github.com/azure/kaito/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog/v2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	ProbePath  = "/healthz"
+	Port5000   = int32(5000)
+	TuningFile = "fine_tuning_api.py"
+)
+
+var (
+	containerPorts = []corev1.ContainerPort{{
+		ContainerPort: Port5000,
+	},
+	}
+
+	livenessProbe = &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Port: intstr.FromInt(5000),
+				Path: ProbePath,
+			},
+		},
+		InitialDelaySeconds: 600, // 10 minutes
+		PeriodSeconds:       10,
+	}
+
+	readinessProbe = &corev1.Probe{
+		ProbeHandler: corev1.ProbeHandler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Port: intstr.FromInt(5000),
+				Path: ProbePath,
+			},
+		},
+		InitialDelaySeconds: 30,
+		PeriodSeconds:       10,
+	}
+
+	tolerations = []corev1.Toleration{
+		{
+			Effect:   corev1.TaintEffectNoSchedule,
+			Operator: corev1.TolerationOpEqual,
+			Key:      resources.GPUString,
+		},
+		{
+			Effect: corev1.TaintEffectNoSchedule,
+			Value:  resources.GPUString,
+			Key:    "sku",
+		},
+	}
+)
+
+func getInstanceGPUCount(sku string) int {
+	gpuConfig, exists := kaitov1alpha1.SupportedGPUConfigs[sku]
+	if !exists {
+		return 1
+	}
+	return gpuConfig.GPUCount
+}
+
+func formatRegistryImagePath(registryName, imageName, imageTag string) string {
+	if imageTag != "" {
+		return fmt.Sprintf("%s/%s:%s", registryName, imageName, imageTag)
+	}
+	return fmt.Sprintf("%s/%s", registryName, imageName)
+}
+
+func GetTuningImageInfo(ctx context.Context, wObj *kaitov1alpha1.Workspace, presetObj *model.PresetParam) string {
+	registryName := os.Getenv("PRESET_REGISTRY_NAME")
+	return formatRegistryImagePath(registryName, "kaito-tuning-"+string(wObj.Tuning.Preset.Name), presetObj.Tag)
+}
+
+func GetDataSrcImageInfo(ctx context.Context, wObj *kaitov1alpha1.Workspace) (string, []corev1.LocalObjectReference) {
+	imagePullSecretRefs := make([]corev1.LocalObjectReference, len(wObj.Tuning.Input.ImagePullSecrets))
+	for i, secretName := range wObj.Tuning.Input.ImagePullSecrets {
+		imagePullSecretRefs[i] = corev1.LocalObjectReference{Name: secretName}
+	}
+	registryName := os.Getenv("PRESET_REGISTRY_NAME")
```

Comment:
Why do we need to get registry name from env?  Can we get the registry from the Image, 
or in the other word, wObj.Tuning.Input.Image should be a valid image url already.

Reply:
Ok - User will specify full url